tests/publiccloud/ipa.pm

```diff
@@ -31,97 +51,58 @@ sub find_secgroup {
-        if (script_run($cmd) == 0) {
+        my $ret      = script_run($cmd);
+        if (defined($ret) && $ret == 0) {
             return $key_name;
         }
     }
     die("Unable to create SSH key on aws with prefix '$prefix'");
 }
 
-sub save_logs {
-    my $ret = script_run("test -d ipa_results");
-    return if (!defined($ret) || $ret != 0);
-
-    my $output = script_output("find ipa_results -type f");
-    for my $file (split(/\n/, $output)) {
-        if ($file =~ m'ipa_results/ec2/ami-[0-9a-z]+/i-[0-9a-z]+/[0-9]{14}\.(log|results)') {
-            upload_logs($file, failok => 1);
-            if ($file =~ /results$/) {
-                parse_extra_log(IPA => $file);
-            }
-        }
-    }
-}
-
-sub run {
+sub ec2_run_ipa {
     my ($self) = @_;
 
-    select_virtio_console();
-
-    die "Public cloud provider isn't supported" unless check_var('PUBLIC_CLOUD_PROVIDER', "EC2");
-
-    if (is_sle) {
-        my $modver = get_required_var('VERSION') =~ s/-SP\d+//gr;
-        add_suseconnect_product('sle-module-public-cloud', $modver);
-    }
-    zypper_call('ar ' . get_required_var('PUBLIC_CLOUD_TOOLS_REPO'));
-    zypper_call('--gpg-auto-import-keys in python3-ipa python3-ipa-tests');
-
-    # WAR install awscli from pip instead of using the package bcs#1095041
-    zypper_call('in gcc python3-pip');
-    if (is_opensuse) {
-        zypper_call('in python3-devel');
-        assert_script_run("pip3 install pycrypto");
-    }
-    assert_script_run("pip3 install awscli");
-    assert_script_run("pip3 install keyring");
+    my $region = get_var('PUBLIC_CLOUD_REGION', 'eu-central-1');
 
     assert_script_run("export AWS_ACCESS_KEY_ID=" . get_required_var('PUBLIC_CLOUD_KEY_ID'));
     assert_script_run("export AWS_SECRET_ACCESS_KEY=" . get_required_var('PUBLIC_CLOUD_KEY_SECRET'));
-    assert_script_run("export AWS_DEFAULT_REGION='eu-central-1'");
+    assert_script_run('export AWS_DEFAULT_REGION="' . $region . '"');
 
     # Create SSH key.
-    my $ssh_key_name = create_ssh_key("openqa_" . time, 'QA_SSH_KEY.pem');
+    my $ssh_key_name = ec2_create_ssh_key("openqa_" . time, 'QA_SSH_KEY.pem');
     $self->{'ipa_ssh_key_name'} = $ssh_key_name;
 
     # Create security group
-    my $secgroup_id = create_secgroup("qa_secgroup");
+    my $secgroup_id = ec2_create_secgroup("qa_secgroup");
     die "Failed on creating security group" unless $secgroup_id;
 
     #Prestart instance, cause IPA might use the wrong security group
     my $ami_id = get_required_var("PUBLIC_CLOUD_IMAGE_ID");
     my ($instance_id)
-      = script_output("aws ec2 run-instances --image-id $ami_id --instance-type t2.large --key-name " . $ssh_key_name . " --security-group-ids '$secgroup_id'")
+      = script_output("aws ec2 run-instances --image-id $ami_id --instance-type '"
+          . get_required_var('PUBLIC_CLOUD_INSTANCE_TYPE', 't2.large')
```

Comment:
main purpose of `get_required_var` function is to fail test when variable not defined. of course you may have case when someone define variable with empty value but it not look like realistic scenario and actually would like to fail also in such case.
So I would suggest two options or `get_var('PUBLIC_CLOUD_INSTANCE_TYPE','t2.large');` or `get_required_var('PUBLIC_CLOUD_INSTANCE_TYPE');` . get_required_var with fallback value does not make sense for me

Reply:
and thanks for a hint I will try to find out get_required_var have fallback option at all :)

Reply:
get_required_var() doesn't have  a fallback. Its a mistake from my side. Thanks for pointing that out.